### ISSUE TITLE
allow not BehaviorSpace reporting after setup

### ISSUE THREAD

**Issue by alan-isaac** [NetLogo/NetLogo#1853]
Please break "Measure runs at every step" into two pieces: "Measure runs after setup" and "Measure runs after each step".
Currently, when a reporter requires simulation data  (so that it makes sense only after one step), there is no convenient way to prevent a `ArgumentTypeException` in the BehaviorSpace output.  This is not student friendly. The most obvious workaround (forcing a step at the end of setup) is pretty ugly, pedagogically speaking.

**Comment by SethTisue** [comment]
The workaround I would normally suggest would be to bulletproof the reporter, so e.g.

    ifelse-value (ticks = 0) ["N/A"] [ ... ]

...that kind of thing

**Comment by alan-isaac** [comment]
That is still an ugly workaround for early pedagogy, imo. Recall that monitors were fixed to deal with the very problem I'm talking about.  The change I propose is intuitive, transparent, low implementation cost, and useful.

**Comment by SethTisue** [comment]
Refresh my memory, it doesn't halt the experiment, right? You just end up a funny-looking entry in the results?

**Comment by alan-isaac** [comment]
You are quite right about the behavior: students just see `ArgumentTypeException` in their CSV file. But these are undergrads in Econ. Such entries are ... scary.

**Comment by memelaser** [comment]
I believe doing what @SethTisue recommended would work, just use NULL and Excel will just load a blank cell.

**Comment by ERSUCC** [comment]
Hi @alan-isaac, this issue has been resolved in NetLogo 6.4.0, which introduces the run metrics condition option for BehaviorSpace experiments. With this new option, you could set a condition such as `ticks > 0` to solve your problem much more smoothly.
